play/plugin/src/main/kotlin/com/github/triplet/gradle/play/tasks/internal/PlayApiService.kt

```diff
@@ -26,8 +25,24 @@ internal abstract class PlayApiService @Inject constructor(
         private val fileOps: FileSystemOperations,
 ) : BuildService<PlayApiService.Params>, OperationCompletionListener, AutoCloseable {
     val publisher by lazy {
-        credentialStream().use {
-            PlayPublisher(it, parameters.appId.get())
+        val useAppDefaultCreds = parameters.useApplicationDefaultCredentials.getOrElse(false)
+        val useExplicitCreds = (parameters.credentials.asFile.orNull != null ||
```

Comment:
This should fail even if the file doesn't exist.
```suggestion
        val useExplicitCreds = (parameters.credentials.present ||
```

Reply:
Do you mean `isPresent`??

Reply:
Yeah sorry I'm forgetting how Kotlin works :sweat_smile:

Reply:
I know the feeling 😛